### PR TITLE
Cleaning setup

### PR DESCRIPTION
**Pull request by VukW**
## Proposed Changes

- Right now `setup.py` places some GANDLF repo files to `site-packages/` of the user's python env. This cleaning helps to keep all gandlf files in `site-packages/GANDLF` so keeping user env tidy + removes other unused parts of `setup.py`
- Fix for #889

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,21 +14,6 @@
...
-class CustomDevelopCommand(develop):
-    def run(self):
-        develop.run(self)
-
-
-class CustomEggInfoCommand(egg_info):
-    def run(self):
-        egg_info.run(self)
-
-
```
Comment (VukW) [inline setup.py:34]: These commands are just not used (I mean, they does not differ from default ones)

File: setup.py [hunk 2]
```diff
@@ -47,25 +29,11 @@ def run(self):
     for item in os.listdir(os.path.dirname(os.path.abspath(__file__)))
     if (os.path.isfile(item) and item.startswith("Dockerfile-"))
 ]
-setup_files = ["setup.py", ".dockerignore", "pyproject.toml", "MANIFEST.in"]
-all_extra_files = dockerfiles + setup_files
-all_extra_files_pathcorrected = [os.path.join("../", item) for item in all_extra_files]
```
Comment (VukW) [inline setup.py:52]: After installement, user receives this files directly in `site-packages/` folder. The reason is using `../` in the path; however, that is caused by the fact these files are located in repo root.

Good practice for building a wheel by `setup.py` (and, actually, the only possible one) is **not to add** these files to the wheel. Only files that are required for module should be added. Moreover, any file that should be added to user's `site-packages/GANDLF` module should be located in module source library also (`./GANDLF/` in our case, not in a repo root).

So, there are two options:
(1) move necessary files to `./GANDLF/` module source folder;
(2) just don't include them to the wheel at all.

I chose the second solution.

File: setup.py [hunk 3]
```diff
@@ -47,25 +29,11 @@ def run(self):
...
-    "data",
-    "grad_clipping",
-    "losses",
-    "metrics",
-    "models",
-    "optimizers",
-    "schedulers",
-    "utils",
+    "testing*",
 ]
```
Comment (VukW) [inline setup.py:37]: They are not a top-level packages anymore (as they are located **inside** top-level GANDLF package). However, the `testing` is a top-level (together with `GANDLF`); thus we need to exclude it from a wheel (right now a wheel installs a `testing` module also)

**Review by sarthakpati — APPROVED** [review]

**Review by sarthakpati — COMMENTED** [review]
Tests failing.

> FAILED testing/test_full.py::test_generic_config_read - TypeError: affine() got an unexpected keyword argument 'degrees'

**Comment by VukW** [comment]
Hi!
Concerning pytest failure - the failed test is actually brought by https://github.com/mlcommons/GaNDLF/pull/889/ - new augmentations. Let's proceed there, restored branch & PR

Concerning linter issues - I'll fix it in this PR

**Comment by szmazurek** [comment]
Hey! 
I agree with your approach @VukW for the setup file. Regarding the augmentations - interesting thing, I did not receive any errors during tests for this pr. Will also take a look, if you find some cause of that earlier please let me know.

**Comment by sarthakpati** [comment]
Thanks, folks!

**Pull request merged**